Backend/TimeTracker.Api/Controllers/ProjectController.cs

```diff
@@ -1,372 +1,429 @@
-        }
-
-        /// <summary>
-        /// Update project details
-        /// </summary>
-        /// <param name="projectDetails">The details of the existing project to update</param>
-        /// <returns>The Id of the project that was updated</returns>
-        [Authorize]
-        [HttpPatch]
-        public async Task<GenericResponseDTO<int>> UpdateProjectDetails(ProjectDetailsDTO projectDetails)
-        {
-            var currentUserId = authHelper.GetCurrentUserId(User);
-
-            var response = new GenericResponseDTO<int>() 
-            { 
-                Success = false
-            };
-
-            Project project = await database.Projects
-                .AsQueryable()
-                .FirstOrDefaultAsync(x => x.Id == projectDetails.ProjectId && x.Teacher.Id == currentUserId && x.ArchivedDate == null);
-
-            if (project == null)
-            {
-                return new GenericResponseDTO<int>()
-                {
-                    Message = "Couldn't find the project",
-                    Success = false
-                };
-            }
-
-            project.Description = projectDetails.Description;
-
-            await database.SaveChangesAsync();
-
-            response.Success = true;
-            response.Data = project.Id;
-
-            return response;
-        }
-
-        /// <summary>
-        /// This endpoint is used to archive or unarchive a project.
-        /// </summary>
-        /// <param name="archiveDetails">The project Id and if the project is being archived or not</param>
-        /// <returns>The archive date that was saved in the database, returns null if it was unarchived.</returns>
-        [Authorize]
-        [HttpPost("Archive")]
-        public async Task<GenericResponseDTO<DateTime?>> ArchiveProject(ArchiveProjectDTO archiveDetails)
-        {
-            var currentUserId = authHelper.GetCurrentUserId(User);
-
-            var response = new GenericResponseDTO<DateTime?>()
-            {
-                Success = true
-            };
-
-            Project archivingProject = await database.Projects
-                .AsQueryable()
-                .Where(x => x.Id == archiveDetails.ProjectId)
-                .Where(x => x.Teacher.Id == currentUserId)
-                .FirstOrDefaultAsync();
-            
-            if(archivingProject == null)
-            {
-                response.Success = false;
-                response.Message = "Could not find the project";
-                return response;
-            }
-
-            if (archiveDetails.Archive)
-            {
-                archivingProject.ArchivedDate = DateTime.UtcNow;
-            }
-            else
-            {
-                archivingProject.ArchivedDate = null;
-            }
-
-            await database.SaveChangesAsync();
-
-            response.Data = archivingProject.ArchivedDate;
-            return response;
-        }
-    }
-}
+﻿using Microsoft.AspNetCore.Authorization;

```

Comment:
Not sure if github is being weird or some formatting happened, but this entire file is showing it is modified.